### ISSUE TITLE
Unable to replicate benchmark results on Eth3D

### ISSUE THREAD

**Issue by Prashant3282** [YoYo000/MVSNet#33]
Hi @YoYo000,

I have been attempting to replicate your results on the lakeside dataset from Eth3D (and again thank you for the preprocessed dataset). However, I am getting some spotty results and I am wondering what parameters you used when running the network and running fusibile as well.

Here are the images of my final reconstruction:

<img width="775" alt="Screen Shot 2019-03-11 at 8 56 02 PM" src="https://user-images.githubusercontent.com/14365715/54174081-f83fbb00-4441-11e9-9c3a-49e929bd468a.png">
<img width="733" alt="Screen Shot 2019-03-11 at 8 55 20 PM" src="https://user-images.githubusercontent.com/14365715/54174082-f8d85180-4441-11e9-88c7-6a38f691012d.png">

As for my setup, I used a max_w of 640 and a max_h of 480. There were some problems in the actual test.py file where if a collection of views had different image sizes, the actual cropping of the images to max_w and max_h failed. I bypassed this by having max_w and max_h be divided by the min width and height of the images (I'm sure this didn't affect anything). 

After running this, I ran fusibile with 0.3 prob_threshold. 

I'm hoping you could let me know where I am going wrong and how to reproduce your benchmarks. 

Thanks,
Prashant

**Comment by YoYo000** [comment]
Hi Prashant, I just fixed the scaling/cropping in the test script, now you may test on ETH3D dataset with  --max_h 480 --max_w 928. 

Meanwhile the --prob_threshold must be tuned if you are not testing on DTU dataset. In my paper I set --prob_threshold to 0.1 for all tanks and temples and ETH3D data. If you are using fusibile for post-processing, I think you could try prob_threshold around ~0.1

**Comment by Prashant3282** [comment]
Thanks @YoYo000, looks much better now, albeit has a decent amount of noise. Probably has to be tuned further in terms of probability threshold.

<img width="793" alt="Screen Shot 2019-03-12 at 1 10 40 AM" src="https://user-images.githubusercontent.com/14365715/54241332-c7ad5f00-44dd-11e9-88fb-6e2796bd035d.png">

Also a few more questions I had for you:

1. In terms of the preprocessed data, did you use COLMAP to do the reconstruction and get the necessary cameras.txt and pair.txt data? I know there was another thread on this earlier but I don't think I necessarily reproduced the files correctly. The extrinsic and intrinsic matrices I think I mostly have figured out from the COLMAP output, but how did you decide depth_min and depth_interval when you were processing the Eth3D dataset?

2. Similarly, for pair.txt I pulled the vis.dat file from running COLMAP using the PMVS method for image undistortion, but the formats are pretty different. COLMAP outputs the visibility list for every image and R-MVSNet expects the top 10 views in terms of some method of scoring you outlined in your paper. Is there some way to translate the visibility list to the pair.txt input R-MVSNet expects?

**Comment by YoYo000** [comment]
@Prashant3282 

1. I use the provided colmap calibration to generate the MVSNet input (camera.txt, images.txt, point3D.txt). The depth min/max are computed using similar function to ComputeDepthRanges() in colmap, and depth sample number is determined by the method mentioned in R-MVSNet supplementary material.

2. The view selection strategy is described in MVSNet paper. Our view selection is different from colmap or cmvs, so the vis.dat cannot be directly converted to the pair.txt file.

**Comment by Prashant3282** [comment]
I assume when you mean provided COLMAP calibration you are referencing the issue "Exporting camera values for use in MVSNet?" Either way, thanks for the response.

Also a few last things:

1. I was wondering if you also had preprocessed datasets for Eth3D training data, as we would like to evaluate the reconstructions on ground-truth depth maps. If you do have that, I would really appreciate getting access to maybe a few training sets.

2. I did a reconstruction with the parameters that you outlined above, and the result is fairly similar, maybe a little less noisy:

<img width="896" alt="Screen Shot 2019-03-13 at 11 15 39 AM" src="https://user-images.githubusercontent.com/14365715/54358876-bd06dd00-4637-11e9-966b-b0087262b5f2.png">

Is this about what MVSNet produced in terms of accuracy of reconstruction? I want to make sure that I have reproduced your benchmark to the fullest extent.

**Comment by YoYo000** [comment]
@Prashant3282 
ETH3D training data for MVSNet:

https://drive.google.com/open?id=1eqcv0Urr-c3Of8RKmTLhXrY_5cZCUvu5

You may use the following code to generate the training list:
```
def gen_eth3d_path(eth3d_data_folder, mode='training'):
    """ generate data paths for eth3d dataset """

    sample_list = []

    data_names = ['delivery_area', 'electro', 'forest', 'playground', 'terrains']


    for data_name in data_names:

        data_folder = os.path.join(eth3d_data_folder, data_name)

        image_folder = os.path.join(data_folder, 'images')
        depth_folder = os.path.join(data_folder, 'depths')
        cam_folder = os.path.join(data_folder, 'cams')

        # index to image name
        index2name = dict()
        dict_file = os.path.join(cam_folder,'index2prefix.txt')
        dict_list = file_io.FileIO(dict_file, mode='r').read().split()
        dict_size = int(dict_list[0])
        for i in range(0, dict_size):
            index = int(dict_list[2 * i + 1])
            name = str(dict_list[2 * i + 2])
            index2name[index] = name

        # image name to depth name 
        name2depth = dict()
        name2depth['images_rig_cam4_undistorted'] = 'images_rig_cam4'
        name2depth['images_rig_cam5_undistorted'] = 'images_rig_cam5'
        name2depth['images_rig_cam6_undistorted'] = 'images_rig_cam6'
        name2depth['images_rig_cam7_undistorted'] = 'images_rig_cam7'

        # cluster
        cluster_file = os.path.join(cam_folder,'pair.txt')
        cluster_list = file_io.FileIO(cluster_file, mode='r').read().split()
        for p in range(0, int(cluster_list[0])):
            paths = []
            # ref image
            ref_index = int(cluster_list[22 * p + 1])
            ref_image_name = index2name[ref_index]
            ref_image_path = os.path.join(image_folder, ref_image_name)
            ref_cam_path = os.path.join(cam_folder, ('%08d_cam.txt' % ref_index))
            paths.append(ref_image_path)
            paths.append(ref_cam_path)
            # view images
            for view in range(FLAGS.view_num - 1):
                view_index = int(cluster_list[22 * p + 2 * view + 3])
                view_image_name = index2name[view_index]
                view_image_path = os.path.join(image_folder, view_image_name)
                view_cam_path = os.path.join(cam_folder, ('%08d_cam.txt' % view_index))
                paths.append(view_image_path)
                paths.append(view_cam_path)
            # depth path
            image_prefix = os.path.split(ref_image_name)[1]
            depth_sub_folder = name2depth[os.path.split(ref_image_name)[0]]
            ref_depth_name = os.path.join(depth_sub_folder, image_prefix)
            ref_depth_name = os.path.splitext(ref_depth_name)[0] + '.pfm'
            depth_image_path = os.path.join(depth_folder, ref_depth_name)
            paths.append(depth_image_path)
            sample_list.append(paths)

    return sample_list
```

**Comment by Prashant3282** [comment]
@YoYo000 

Hey, sorry to bring this issue back up, but is it possible for you to go a bit more in depth on how you do the view selection? I am currently trying to recreate the pair.txt file from different datasets to test MVSNet, but I am having trouble with it. How exactly do you use the COLMAP files to generate that file? In most of my reconstructions, the mean track length is about 3 or 4, so how is it possible to generate the top 10 highest scores for a certain image?

**Comment by YoYo000** [comment]
@Prashant3282 I am working on generating mvsnet inputs from colmap sfm output. Hopefully I can finish this week.

**Comment by Prashant3282** [comment]
@YoYo000 Great! Thank you for helping me out with this. Please keep me posted.

**Comment by YoYo000** [comment]
@Prashant3282 I just uploaded a script for the input conversion.

**Comment by Prashant3282** [comment]
@YoYo000 Thanks! This is a big help.

About the training sets, how exactly did you generate the depth maps for MVSNet to train on? For example with the DTU dataset? We want to do some similar training with our model but are not sure how to generate those training depth maps.

**Comment by Luxcy** [comment]
@YoYo000  Thanks!
Why the shape of depth_image is (160*128)?how to generate?
